File: Paint_FrontEnd/src/app/Shapes/square.ts
import { IShape } from "./ishape";

export class Square extends IShape {
    
    private _sideLength: number;

    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean, sideLength: number){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._sideLength = sideLength;
        this.type = "square";
    }

    public get sideLength(): number {
        return this._sideLength;
    }
    public set sideLength(value: number) {
        this._sideLength = value;
    }
}

File: Paint_FrontEnd/src/main/java/com/paint/paint_backend/Shapes/Square.java
package com.paint.paint_backend.Shapes;

public class Square extends IShape{
    private int _sideLength;

    Square(int x, int y, String stroke, double strokeWidth, String fill, int rotate, boolean draggable, String id, boolean update, boolean deleteflag, int sideLength){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._sideLength = sideLength;
        super.set_type("square");
    }


    public int get_sideLength() {
        return _sideLength;
    }

    public void set_sideLength(int _sideLength) {
        this._sideLength = _sideLength;
    }
}
